libs/utils/src/cast_vote_records.ts

```diff
@@ -118,6 +129,35 @@ export function convertCastVoteRecordVotesToTabulationVotes(
   return votes;
 }
 
+/**
+ * Converts the mark data of the original snapshot into the simple
+ * dictionary of contest ids to mark scores that VxAdmin uses
+ * internally as a basis for adjudicating marks.
+ */
+export function convertCastVoteRecordMarkMetricsToMarkScores(
+  originalSnapshot: CVR.CVRSnapshot
+): Tabulation.MarkScores {
+  const markScores: Tabulation.MarkScores = {};
+  for (const cvrContest of originalSnapshot.CVRContest) {
+    const contestMarkScores: Record<ContestOptionId, Tabulation.MarkScore> = {};
+    for (const cvrContestSelection of cvrContest.CVRContestSelection) {
+      // We assume every contest selection has only one selection position
+      assert(cvrContestSelection.SelectionPosition.length === 1);
+      const selectionPosition = cvrContestSelection.SelectionPosition[0];
+      assert(selectionPosition);
+      const markMetric = safeParseNumber(selectionPosition.MarkMetricValue);
+      /* istanbul ignore next - @preserve */
+      if (markMetric.isErr()) {
+        throw new Error('unable to interpret mark metric value as number');
+      }
+      contestMarkScores[cvrContestSelection.ContestSelectionId] =
```

Comment:
You could simplify this a bit.
```suggestion
      const markMetric = safeParseNumber(selectionPosition.MarkMetricValue).expect('unable to interpret mark metric value as number');
      contestMarkScores[cvrContestSelection.ContestSelectionId] = markMetric;
```